routes/user.go

```diff
@@ -928,6 +916,55 @@ func (fes *APIServer) augmentProfileEntry(
 	return profileEntryResponse
 }
 
+func (fes *APIServer) _getProfilePictureForPublicKey(publicKey []byte) (string, error) {
+	defaultProfilePicture := "/assets/img/default_profile_pic.png"
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		return defaultProfilePicture, fmt.Errorf("_getProfilePictureforPublicKey: Error getting utxoView: %v", err)
+	}
+
+	profileEntry := utxoView.GetProfileEntryForPublicKey(publicKey)
+	if profileEntry == nil {
+		return defaultProfilePicture, nil
+	}
+	profilePic := string(profileEntry.ProfilePic)
+	if !strings.HasPrefix(profilePic, "data:image/") && !lib.ProfilePicRegex.Match([]byte(profilePic)) {
+		return defaultProfilePicture, nil
+	}
+	return profilePic, nil
+}
+
+type GetSingleProfilePictureRequest struct {
+	PublicKeyBase58Check string `safeForLogging:"true"`
+}
+
+type GetSingleProfilePictureResponse struct {
+	ProfilePic string
+}
+
+func (fes *APIServer) GetSingleProfilePicture(ww http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	publicKeyBase58Check, publicKeyBase58CheckExists := vars["publicKeyBase58Check"]
+	if !publicKeyBase58CheckExists {
+		_AddBadRequestError(ww, fmt.Sprintf("GetSingleProfilePicture: Missing public key base 58 check"))
+		return
+	}
+	publicKeyBytes, _, err := lib.Base58CheckDecode(publicKeyBase58Check)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetSingleProfilePicture: Problem decoding user public key: %v", err))
+		return
+	}
+	// We ignore errors here and simply return
+	profilePicture, _ := fes._getProfilePictureForPublicKey(publicKeyBytes)
+
+	if err = json.NewEncoder(ww).Encode(GetSingleProfilePictureResponse{
```

Comment:
We should just serve the image itself as bytes so browsers recognize the MIME type and we get free client and edge caching

Reply:
If no image, return a 404 and let the frontend use the default_profile_pic